Auth0

jaydp17 opened the pull request:
- [x] Integrates Auth0 for login/signup
- [x] Migrate `My Trips` Page over to new API

fcmatteo commented on src/scenes/registrations/actions.js:
```diff
@@ -29,18 +31,44 @@ export const registrationFailed = error_payload => {
+        try {
+          const auth0Response = await axios
+            .post(`${serverBaseURL}/users/login`, { username: email, password: password })
+            .catch(error => {
```
In theory this catch is not necessary. It will be catched by the `try/catch`. And actually as you are using this catch, the error will not bubble up, and this function will continue executing.
The same with the catch below (line 55 https://github.com/PleaseDotCom/frontend/pull/202/files#diff-528b1a1716ba6d7be34b6bc600d9bb89R55 ). As an example, after the catch in the line 55, it will try to get `user.data.accessToken` and as `user.data` will be `undefined` the app will crash.

jaydp17 replied:
@fcmatteo agreed, I've refactored it to have 1 try/catch that catches all the errors and shows it in the UI

fcmatteo commented on src/scenes/registrations/containers/registrations_container.js:
```diff
@@ -16,6 +19,9 @@ class RegistrationsContainer extends Component {
     };
     this.handleInputChange = this.handleInputChange.bind(this);
     this.isInputInvalid = this.isInputInvalid.bind(this);
+    if (localStorage.getItem(`please-${env}-session`)) {
```
Maybe `please-${env}-session` and all of the local storage variables could be in a file as "constants".

jaydp17 replied:
@fcmatteo that's a valid point, I thought I moved every reference of this inside `user-session.js` but missed this one.
I've changed it to use `getSession()` from `user-session.js`

fcmatteo commented on src/scenes/sessions/containers/sessions_container.js:
```diff
@@ -17,6 +17,9 @@ class SessionsContainer extends Component {
     this.handleInputChange = this.handleInputChange.bind(this);
     this.isInputInvalid = this.isInputInvalid.bind(this);
     this.isLoginError = this.isLoginError.bind(this);
+    if (localStorage.getItem('please-development-session')) {
```
This is only wanted on development?

jaydp17 replied:
@fcmatteo great catch.
@dm0lz I've changed it to `please-${env}-session` inside `user-session.js`, let me know if this was meant only for development

fcmatteo commented on src/shared_components/I18nText.js:
```diff
@@ -0,0 +1,17 @@
+ * Later we can use this component to render the user selected language
+ */
+function I18nText({ data }) {
+  return <React.Fragment>{data['en-us']}</React.Fragment>;
```
I like the idea of this component :) Will do the same on mobile.

fcmatteo commented on src/styled_scenes/Account/Profile/index.js:
```diff
@@ -132,7 +132,7 @@ class AccountProfileScene extends Component {
             )}
           </HorizontalSpan>
-          <HorizontalSpan>
+          {/*<HorizontalSpan>
```
Why is this commented?

jaydp17 replied:
@fcmatteo I have no idea 🤷‍♂️
As login/signup part was worked on by Olivier, maybe he knows.
@dm0lz ?

fcmatteo commented on src/styled_scenes/Account/components/UserBasicInfo.js:
```diff
@@ -121,8 +121,8 @@ const UserBasicInfo = ({ user_profile: user = {}, match }) => {
-
-          <div style={{ cursor: 'pointer' }} onClick={logout}>
+            */}
+          <div style={{ cursor: 'pointer' }} onClick={() => logOut()}>
```
I don't understand this change. It should work with just `onClick={logOut}` and it's not the best to have arrow functions in render, as it will generate a new function on each render.

jaydp17 replied:
@fcmatteo yes, we don't need the additional arrow function.
We should probably have a linter rule that warns us about this.
Anyways for now I've removed the arrow function.

fcmatteo commented on src/scenes/account/actions.js:
```diff
@@ -34,26 +35,16 @@ export const edit_user_error_raised = error => {
+      console.log(error);
+      //dispatch(setLoginError({code: error.response.status, message: error.response.data.error_description}));
+    });
+    dispatch(user_profile_fetched({ user_profile: user.data }));
```
Here we have the same problem with the catch (the function will continue executing). And even if there is not an error, the action is dispatched before the promise fulfills (an `await` is missing).

jaydp17 replied:
@fcmatteo fixed

jaydp17 commented:
Disclaimer: A lot of code in this PR was worked on by @dm0lz, I don't have a lot of context on it, but I've modified it to the best of my knowledge.
@dm0lz If I changed something that shouldn't have been, do let me know.

@fcmatteo this PR should not use the old API `https://please-api.herokuapp.com/users/signup` how is that you're getting the error?

The error with the new API is legit, I never tested with that, what's happening is because in `lib/axios.js` we had an interceptor that said if the user is logged in with Parse attach the `X-Parse-Session-Token` to the request and if the user is logged in with Auth0 attach `Authorization` header, what happened here is that it attached both, which should not happen.

Thus, as a simpler fix, I've removed the Parse related code from the axios interceptor.

Also, @fcmatteo thanks for the great review 👏
Feels like going forward all my PRs are landing on you to review 😁

jaydp17 commented on src/scenes/sessions/components/login_form_component.jsx:
```diff
@@ -119,6 +119,9 @@ export default class LoginFormComponent extends Component {
                       </Button>
                     </Segment>
                   </Form>
+
```
@dm0lz should we just remove the MetaMask/Ledger code as we can always get that back from the `demo` branch?

jaydp17 commented:
@andrew-t-james I'd suggest we first merge this PR and then create a separate PR for service creation changes.
@dm0lz @fcmatteo @vlddlv Please have a look so that we can merge this.
Once @andrew-t-james confirms, I'll go ahead and remove the latest commits by him, so we can review those changes separately
